src/main/java/com/redhat/cloud/policies/app/rest/PolicyCrudService.java

```diff
@@ -356,27 +456,72 @@ public Response deletePolicy(@PathParam("id") UUID policyId) {
     }
     Policy policy = Policy.findById(user.getAccount(), policyId);
 
-    ResponseBuilder builder ;
+    ResponseBuilder builder = Response.ok();
     if (policy==null) {
       builder = Response.status(Response.Status.NOT_FOUND);
     } else {
-      policy.delete(policy);
+      boolean deletedOnEngine = false;
       try {
         engine.deleteTrigger(policy.id, user.getAccount());
-        builder = Response.ok(policy);
+        deletedOnEngine = true;
       } catch (NotFoundException nfe) {
         // Engine does not have it - we can delete anyway
-        builder = Response.ok(policy);
+        deletedOnEngine = true;
       } catch (Exception e) {
-        e.printStackTrace();  // TODO: Customise this generated block
+        log.warning("Deletion on engine failed because of " + e.getMessage());
         builder = Response.serverError().entity(new Msg(e.getMessage()));
       }
+      if (deletedOnEngine) {
+        policy.delete(policy);
+        builder = Response.ok(policy);
+      }
     }
 
     return builder.build();
 
   }
 
+  @Operation(summary = "Delete policies for a customer by the ids passed in the body. Result will be a list of deleted UUIDs")
+  @APIResponse(responseCode = "403", description = "Individual permissions missing to complete action")
+  @APIResponse(responseCode = "200", description = "Policies deleted",
+      content = @Content(schema = @Schema(type = SchemaType.ARRAY, implementation = UUID.class)))
+  @DELETE
+  @Path("/ids")
+  @Transactional
+  public Response deletePolicies(List<UUID> uuids) {
+
+    if (!user.canWriteAll()) {
+       return Response.status(Response.Status.FORBIDDEN).entity(new Msg("Missing permissions to delete policy")).build();
+    }
+
+    List<UUID> deleted = new ArrayList<>(uuids.size());
+
+    for (UUID uuid : uuids) {
+      Policy policy = Policy.findById(user.getAccount(), uuid);
+      if (policy == null ) {
+        // Nothing to do for us
+        deleted.add(uuid);
+      } else {
+        boolean deletedOnEngine = false;
+        try {
+          engine.deleteTrigger(policy.id, user.getAccount());
+          deletedOnEngine = true;
+        } catch (NotFoundException nfe) {
+          // Engine does not have it - we can delete anyway
+          deletedOnEngine = true;
+        } catch (Exception e) {
+          log.warning("Deletion on engine failed because of " + e.getMessage());
+        }
+        if (deletedOnEngine) {
+          policy.delete();
```

Comment:
I'm not really sure about this operation, at this point, we have already deleted the policy from the engine, lets say it fails on the last (out of 10) policy (for whatever reason), then we won't delete the policy from the backend but it will be gone from the engine. I think we just should catch this exception and skip the `deleted.add(uuid)`

Reply:
If it fails for whatever reason to delete on the engine, `deletedOnEngine` is false and `policy.delete()` would not kick in.
Perhaps you mean something else?

Reply:
I mean if it fails in `policy.delete`, although maybe not that likely.